apps/system/js/advanced_telemetry.js

```diff
@@ -379,13 +377,17 @@
         self.startRetryBatch();
       }
 
-      request.send({
-        timeout: AT.REPORT_TIMEOUT,
-        onload: onload,
-        onerror: retry,
-        onabort: retry,
-        ontimeout: retry
-      });
+      if (navigator.onLine) {
+        request.send({
+          timeout: AT.REPORT_TIMEOUT,
+          onload: onload,
+          onerror: retry,
+          onabort: retry,
+          ontimeout: retry
+        });
+      } else {
```

Comment:
The changes look good. I can see how they solve the problem, and the code is much cleaner. My only constructive comment is the readability would be improved by checking 'navigator.onLine' at the beginning of the 'send' function. (Jan's 'bailout' style is rubbing off on me).